File: storage/data_transpiled/at_coder/abc034/C/4563469.java
private static int Mod_Combination ( int n , int k , int mod ) {
  long p = 1 , q = 1 ;
  for ( int i = n - k + 1 ;
  i <= n ;
  i ++ ) {
    p = ( p * i ) % mod ;
  }
  for ( int i = 2 ;
  i <= k ;
  i ++ ) {
    q = ( q * i ) % mod ;
  }
  return ( int ) ( p * Mod_Inverse ( q , mod ) % mod ) ;
}

File: storage/data_original/at_coder/abc034/C/4563469.py
def Mod_Combination(n, k, mod):
    p, q = 1, 1
    for i in range(n-k+1, n+1):
        p = (p * i) % mod
    for i in range(2, k+1):
        q = (q * i) % mod
    return int(p * Mod_Inverse(q, mod) % mod)

def Ext_Gcd(a, b):
    if b == 0:
        return a, 1, 0
    else:
        d, x, y = Ext_Gcd(b, a % b)
        x -= (a // b) * y
        return d, y, x

def Mod_Inverse(a, mod):
    return Ext_Gcd(a, mod)[1] % mod

x, y = map(int, input().split())
print(Mod_Combination(x + y - 2, x - 1, 10**9 + 7))